src/server.py

```diff
@@ -139,6 +132,7 @@ def parse_request(request):
 
 
 def assemble_response(request):
```

Comment:
I like this level of encapsulation.  This is a nice, clean looking function that basically encompasses everything your server does.  It even looks testable.  Well done!